File: C++/2185CountingWordsWithGivenPrefix.cpp
//
//  2185CountingWordsWithGivenPrefix.cpp
//  Testing C++
//
//  Created by Cem Beyenal on 1/9/25.
//
#include <iostream> 
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int prefixCount(vector<string>& words, string pref) {
        int count = 0;
        for(int i = 0; i < words.size(); i++)
        {
            bool isPrefix = true;
            for (int j = 0; j < pref.size(); j++)
            {
                if (words[i][j] != pref[j])
                {
                    isPrefix = false;
                    break;
                }
            }
            if (isPrefix)
            {
                count++;
            }
        }
        return count;
    }
};

File: Python/2185CountingWordsWithGivenPrefix.py
class Solution(object):
    def prefixCount(self, words, pref):
        """
        :type words: List[str]
        :type pref: str
        :rtype: int
        """
        count = 0
        for i in range(len(words)):
            isPrefix = True
            for j in range(len(pref)):
                if j >= len(words[i]) or words[i][j] != pref[j]:
                    isPrefix = False
                    break
            if isPrefix:
                count += 1
        return count


# Better solution using python functions
class Solution1(object):
    def prefixCount1(self, words, pref):
        count = 0

        for word in words:
            if word.startswith(pref):
                count += 1

        return count